// components/searchByIngredients/index.js
import React, { Component } from 'react'
// import {Link} from 'react-router'
import {connect} from 'react-redux'

import {fetchRecipes} from 'actions'

// import './index.css'
import './index.sass'

class SearchByIngredients extends Component {
  constructor(props) {
    super(props)
    this.state = {
      allowedIngredient: '',
      excludedIngredient: ''
    }

    this.handleAllowedIngredientChange = this.handleAllowedIngredientChange.bind(this)
    this.handleExcludedIngredientChange = this.handleExcludedIngredientChange.bind(this)
    this.handleClick = this.handleClick.bind(this)    
  }

  handleClick (event) {
    this.props.fetchRecipes(this.state.allowedIngredient, this.state.excludedIngredient)
  }

  handleAllowedIngredientChange (event) {
    this.setState({
      allowedIngredient: event.target.value
    })
  }

  handleExcludedIngredientChange (event) {
    this.setState({
      excludedIngredient: event.target.value
    })
  }

  render () {
    return (
      <div className='search-ingredients'>
        <div className='search-ingredients__enter'>
          <h3 className='search-ingredients__title'>Search by ingredients</h3>
          <div>
            <div className='search-ingredients__block'>
              <input
                onChange={this.handleAllowedIngredientChange}
                type='text'
                className='search-ingredients__input'
                placeholder='eggs, tomato paste'
              />
            </div>
            <div className='search-ingredients__block'>
              <input
                onChange={this.handleExcludedIngredientChange}
                type='text'
                className='search-ingredients__input'
                placeholder='eggs, tomato paste'
              />
            </div>
            <div>
              <button 
                className='search-ingredients__btn'
                onClick={() => this.handleClick()}
              >
                Start
              </button>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = {
  fetchRecipes
}

export default connect(null, mapDispatchToProps)(SearchByIngredients)
