// containers/recipe/index.js
import React, {Component} from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router'

import { fetchRecipeById, addRecipeToBasket } from 'actions'
import { getRecipe } from 'selectors'

import BasketCart from 'components/basketCart'
import Header from 'components/header'
import Footer from 'components/footer'

// import './index.css'
import './index.sass'

class Recipe extends Component {
  componentDidMount () {
    this.props.fetchRecipeById(this.props.params.id)
  }

  renderList () {
    const {recipe} = this.props
    
    return recipe.ingredientLines.map((value, key) => (
      <div className='ingredient-item' key={key}>
        <div 
          className='ingredient-element'
        >
          {value}
        </div>
      </div>
    ))
  }

  renderContent () {
    const {recipe} = this.props
    return (
      <div className='recipe-info'>
        <div className='recipe-name'>
          <div>
            <h4>Recipe name</h4>
          </div>
          <div>
            <h4>
              {recipe.name}
            </h4>
          </div>
        </div>
        <div className='recipe-body'>
          <div className='recipe-block-image'>
            <div>
              <img
                src={recipe.images[0].hostedMediumUrl}
                alt={recipe.name}
                className='recipe-image'
              />
            </div>
            <div className='recipe-servings'>
              <div className='recipe-servings__name'>
                <span>Servings:</span>
              </div>
              <div>
                <span>{recipe.numberOfServings}</span>
              </div>
            </div>
            <div className='recipe-totalTime'>
              <div className='recipe-totalTime__name'>
                <span>Cooking time:</span>
              </div>
              <div>
                <span>{recipe.totalTime}</span>
              </div>
            </div>
          </div>
          <div className='recipe-ingredients'>
            <div className='ingredient-list'>
              <span 
                className='ingredient-name'
              >
                Ingredients
              </span>
              {this.renderList()}
            </div>
            <div className='recipe-category'>
              {
                recipe.attributes.course &&
                <div className='recipe-category__list'>
                  <h4 className='recipe-category__title'>Course</h4>
                  <span>
                      {recipe.attributes.course.map((value, index) => (
                        <div key={index} className='recipe-category__element'>
                          <span>{value}</span>
                        </div>
                      ))}
                    </span>
                </div>
              }
              {
                recipe.attributes.cuisine &&
                <div className='recipe-category__list'>
                  <h4 className='recipe-category__title'>Cuisine</h4>
                  <span>
                      {recipe.attributes.cuisine.map((value, index) => (
                        <div key={index} className='recipe-category__element'>
                          <span>{value}</span>
                        </div>
                      ))}
                    </span>
                </div>
              }
            </div>
          </div>
        </div>
        <div className='recipe-footer'>
          <a 
            href={recipe.source.sourceRecipeUrl}
            className='recipe-footer__link'
            target="_blank"
          >
            Read directions
          </a>
        </div>
      </div>
    )
  }

  renderSidebar () {
    const {recipe, addRecipeToBasket} = this.props
    return (
      <div className='recipe-sidebar'>
        <BasketCart />
        <div className='recipe-sidebar__buttons'>
          <Link
            to='/'
            className='recipe-sidebar__btn sidebar-btn--backToHome'
          >
            Back to home
          </Link>
          <button
            type='button'
            className='recipe-sidebar__btn sidebar-btn--addToBasket'
            onClick={() => addRecipeToBasket(recipe.id)}
          >
            Add to basket
          </button>
          </div>
      </div>
    )
  }

  render () {
    const {recipe} = this.props
    return (
      <div>
        <Header />
        <div className='recipe-container'>
          <div className='recipe-section'>
            <div className='recipe-row'>
              <div className='recipe-content'>
                  {recipe && this.renderContent()}
              </div>
              <div className='recipe-nav'>
                  {recipe && this.renderSidebar()}
              </div>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    )
  }
}

const mapStateToProps = state => {
  return {
    recipe: getRecipe(state, state.recipePage.id)
  }
}

const mapDispatchToProps = {
  fetchRecipeById,
  addRecipeToBasket
}

export default connect(mapStateToProps, mapDispatchToProps)(Recipe)
