// api/mockRecipes.js
export default [
    {
        id: "1",
        uri: "http://www.edamam.com/ontologies/edamam.owl#recipe_114fe9273fd61b08c61f377e98b375bf",
        label: "Horseradish Cream",
        image: "https://www.edamam.com/web-img/f7e/f7ed40148b711b6c207cdb42c2f0713d.jpg",
        source: "Leite's Culinaria",
        url: "http://leitesculinaria.com/95386/recipes-horseradish-cream.html",
        shareAs: "http://www.edamam.com/recipe/horseradish-cream-114fe9273fd61b08c61f377e98b375bf/cream",
        yield: 4.0,
        ingredientLines: [
            "1/4 to 1/2 cup grated fresh horseradish, or more to taste",
            "1 cup crème fraîche or sour cream",
            "1 tablespoon freshly squeezed lemon juice",
            "1 tablespoon chopped chives (optional)",
            "Kosher salt, to taste"
        ],
        ingredients: [
        {
            text: "1/4 to 1/2 cup grated fresh horseradish, or more to taste",
            weight: 88.72058868408203
        },
        {
            text: "1 cup crème fraîche or sour cream",
            weight: 224.0
        },
        {
            text: "1 tablespoon freshly squeezed lemon juice",
            weight: 14.0
        },
        {
            text: "Kosher salt, to taste",
            weight: 1.9603235321044923
        }
        ],
        calories: 477.9858825683594,
        totalWeight: 327.44749854036957,
    },
    {
        id: "2",
        uri: "http://www.edamam.com/ontologies/edamam.owl#recipe_beed5bf23fe7f26d3f5a02bb6c461953",
        label: "Maple Cream",
        image: "https://www.edamam.com/web-img/535/535c23f8c7c7d9d658d3d474ce22ea19.jpg",
        source: "Food52",
        url: "http://food52.com/recipes/26640-maple-cream",
        shareAs: "http://www.edamam.com/recipe/maple-cream-beed5bf23fe7f26d3f5a02bb6c461953/cream",
        yield: 8.0,
        ingredientLines: [
            "1 quart 100% pure maple syrup (Grade A Light Amber is ideal)",
            "1/4 teaspoon cream (or substitute oil to keep it dairy-free)"
        ],
        ingredients: [
            {
                text: "1 quart 100% pure maple syrup (Grade A Light Amber is ideal)",
                weight: 1260.0
            },
            {
                text: "1/4 teaspoon cream (or substitute oil to keep it dairy-free)",
                weight: 1.2595723867416382
            }
        ],
        calories: 3280.3455247342586,
        totalWeight: 1261.2595723867416,
    }
]
